feat: add digital pet game lab

naomi-lgbt opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or Gitpod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #XXXXX

<!-- Feel free to add any additional description of changes below this line -->

naomi-lgbt committed:
feat: scaffold the project

naomi-lgbt committed:
chore: forgot the seed oopsie

jdwilkin4 reviewed (commented):
Before we have the cohort start working on these, we will need to update the titles and dashed names to match the convention laid out for the other workshops and labs.


The workshop dashed names should say workshop- and the labs should say lab-

so in this case lab-digital-pet-game would work 👍🏾

a2937 committed:
Merge branch 'freeCodeCamp:main' into feat/digital-pet-game

a2937 committed:
rename the project

ElectricMolasses committed:
First pass of user stories.
Corrected a couple bugs that existed in the example solution while
testing.
- `Hunger` bar was coloured inversely to how it should be
- `Enegy` was able to exceed `100`

ElectricMolasses committed:
Merge pull request #1 from nhcarrigan/feat/digital-pet-game_initial-user-stories

Initial User Story Pass

a2937 committed:
move the lab

a2937 committed:
Merge branch 'main' into feat/digital-pet-game

ElectricMolasses committed:
Add digital pet game hints

Updated the user stories to reflect additional requirements for testing.
Added hint text, no tests yet.

a2937 committed:
Apply suggestions from code review

Co-authored-by: Jessica Wilkins  <67210629+jdwilkin4@users.noreply.github.com>

a2937 committed:
Merge branch 'freeCodeCamp:main' into feat/digital-pet-game

a2937 committed:
get tests to function correcrtly in pet game

ElectricMolasses committed:
Digital Pet Game Tests

Initial pass nearly complete, lingering error regarding the change of
value after excessive idle time. I've tried a few approaches and can't
get it to organically exceed what appears two be two ticks.

All other tests seems to be working as expected so far, but testing the
state of how the application responds to button clicks needs to be made
more thorough. This isn't difficult, but is cleaned with the ability to
simulate idle time.

ElectricMolasses committed:
Ready for Review

- Initial pass of tests completed, currently the `clock.tickAsync` tests
  do not function on my machine, but do function on other machines, more
  testing required here.
- Removed numbering from the user stories and flattened down to two `ul`
  levels deep.

a2937 committed:
remove merge conflict

a2937 committed:
move block to front end v9

a2937 committed:
hopefully not damaged submodule

a2937 committed:
confused.

ElectricMolasses committed:
Correct failing tests

All tests began failing as none of the helper functions in the
`before-all` block were accessible. I believe this was the result of the
challenge type being changed, DOM challenges are supposed to function
this way, but the docs on `how-to-work-on-coding-challenges` state that
other challenges need to have variables that are reused assigned to
`globalThis`.

Adding them all to `globalThis` resolved the bulk of the issues. There
were some lingering case and id issues in the sample solution, but I
updated that with my own sample code I was testing against in the
client.

ElectricMolasses committed:
Mitigate scope conflict with learner code

Added underscores to my test helper functions to avoid potential already
defined errors with the campers own code

a2937 committed:
remove curriculum.json

jdwilkin4 reviewed (commented):
Just started reviewing this and the UI looks off here. 
It shows a duplicate of the user stories in the editor

<img width="1506" height="740" alt="Screenshot 2026-01-08 at 8 44 18 AM" src="https://github.com/user-attachments/assets/d1eae0d8-c9fc-4b4f-b525-7753f4685a33" />

Also, it looks like this branch hasn't been updated in a while. 
So it would be good to do that.

a2937 committed:
Merge branch 'freeCodeCamp:main' into feat/digital-pet-game

a2937 committed:
move digital pet came intro before review

ElectricMolasses committed:
Remove inline instructions from lab-digital-pet-game

Applying suggestion that removes inline instructions from the digital pet game lab and moves them into the dedicated instruction area.

Co-authored-by: Dario <105294544+Dario-DC@users.noreply.github.com>

ElectricMolasses committed:
Node 24 update

Merge remote-tracking branch 'upstreamm/main' into feat/digital-pet-game

majestic-owl448 commented on client/src/pages/learn/front-end-development-libraries-v9/build-a-digital-pet-game/index.md:
can you please delete this file? there is been a new recent change and we are not using them anymore

Dario-DC commented on curriculum/challenges/english/blocks/lab-digital-pet-game/68c362b379059c388d3874f2.md:
```diff
@@ -0,0 +1,1434 @@
+
+**User Stories:**
+
+The application should have two views. The requirements for each view are encompassed by the rest of the user stories:
```
I haven't started yet to review this PR but I noticed that this doesn't follow what we are doing for other labs. User stories should be written as an ordered list.

a2937 committed:
Merge branch 'freeCodeCamp:main' into feat/digital-pet-game

a2937 committed:
Correct user story spelling

Co-authored-by: Sem Bauke <sem@freecodecamp.org>

a2937 committed:
remove unnecessary stuff

Sembauke commented:
In the description, EXCITED is Happiness > 80 and Energy < 70, but solution code uses energy > 70.

Sembauke commented on curriculum/challenges/english/blocks/lab-digital-pet-game/68c362b379059c388d3874f2.md:
````diff
@@ -0,0 +1,1434 @@
+Each `.stat-value` number should be within the range `[0,100]`, inclusive.
+
+```js
+const statValueEls = document.querySelectorAll('stat-value');
````
```suggestion
const statValueEls = document.querySelectorAll('.stat-value');
```

Sembauke commented on curriculum/challenges/english/blocks/lab-digital-pet-game/68c362b379059c388d3874f2.md:
````diff
@@ -0,0 +1,1434 @@
+Each `.stat-value` element should contain one number.
+
+```js
+const statValueEls = document.querySelectorAll('stat-value');
````
```suggestion
const statValueEls = document.querySelectorAll('.stat-value');
```

Ksound22 committed:
Merge branch 'main' into feat/digital-pet-game

Ksound22 committed:
linter happy?

Ksound22 committed:
rewrite tests

Ksound22 committed:
Merge branch 'main' into feat/digital-pet-game

Ksound22 committed:
a better intro text

Ksound22 committed:
correct excited mood condition in description

Ksound22 committed:
Merge branch 'main' into feat/digital-pet-game
